### DISCUSSION TITLE
Unable to export flows

### DISCUSSION THREAD

**Discussion by Captain-E** [akvorado/akvorado#853] [Q&A] [answered]
Hi Vincent.
Thank you for this amazing flow collector project.

Few days ago, i configured NetFlow on my ASA firewalls. But the flows weren't exported.
Akvorado is working fine with my switches but it doesn't accept flows from ASA firewalls.

As you can see, ASA is sending flows to the analyzer: 
![image](https://github.com/akvorado/akvorado/assets/76643470/21c50a04-a357-4670-b156-c5de1db63732)

But these flows are not exported and processed. So i made some troubleshooting:
![image](https://github.com/akvorado/akvorado/assets/76643470/3601ea2a-63be-4b03-bcca-a54e759af16e)
![image](https://github.com/akvorado/akvorado/assets/76643470/58f1dfad-9d7d-44ea-9c82-bf0d26c682c2)

It seems like akvorado was unable to get ASA interface parameters but i configured snmp community on ASA and allowed snmp from akvorado host.

PS: ASA working fine with Elastiflow and Elastiflow is processing flows without any issues. Do you have any ideas.
**ASA snmp config:** snmp-server host <asa_interface> <akvorado_host> community ***** version 2c

**Comment by vincentbernat** [comment]
I don't think the problem is with SNMP as you only have one error. However, you have many errors about template missing. Check if you see them on your capture with Wireshark.

**Reply by Captain-E** [reply]
"template missing" error counters were increasing until snmp_poller_error was displayed. Then "template missing" counters stopped. That's why i thought it was snmp issue.
I captured netflow packets using tcpdump and analyzed it using wireshark. It says warning "no template found". Does this means that Netfow is not working correctly on ASA side?
![image](https://github.com/akvorado/akvorado/assets/76643470/ee91c393-4a00-4e85-ac15-04dc4a1ab4cf)

![image](https://github.com/akvorado/akvorado/assets/76643470/e7009cf1-a3b3-42fc-a2fb-b8f6f9dfcb8f)

**Reply by vincentbernat** [reply]
Maybe your ASA device is sending the template infrequently. You should run the capture long enough to have the template (so Wireshark can decode) and an SNMP error. Then, you can check the SNMP request sent and the answer you get. If you get an SNMP error about "noSuchInstance", you can then check with "snmpwalk" if the index is correct. The index should be the one sent in Netflow. Since you have only one SNMP error, maybe the interface index is missing. That may be something you can configure.

**Reply by Captain-E** [reply]
Hi. Thanks for staying in touch :D

I've changed template timeout using "flow-export template timeout-rate 1" command. 
After that my ASA fw started sending template and i captured flow data.
![image](https://github.com/akvorado/akvorado/assets/76643470/40ec5330-63b4-4910-9ea1-bcc8786627d4)
ASA config: 
flow-export delay flow-create 30
flow-export active refresh-interval 1
flow-export template timeout-rate 1

However flows are still not displayed on the akvorado web interface. As you may have noticed, I captured interface id and flow data, so i assume snmp is not problem. Therefore i restarted all the dockers containers and searched for error.
template not found error stopped increasing but sampling rate missing error keeps increasing. Could it be the reason why flows aren't displayed?
![image](https://github.com/akvorado/akvorado/assets/76643470/2e154209-c559-4342-8303-88e1a4a58659)
![image](https://github.com/akvorado/akvorado/assets/76643470/b83c2e94-3ce7-4018-a7bb-26e20b756eef)

**Reply by vincentbernat** [reply]
Yes, check https://demo.akvorado.net/docs/troubleshooting#no-packets-exported.

**Reply by Captain-E** [reply]
Hi.
i apologize for asking many things

Since i couldn't find any configuration to set sampling rate on ASA, i decided to add "default-sampling-rate: 10" option and it fixed sampling rate missing error. Flows are now displayed on web interface and i'm glad that it works. But there's another issue. Flow graphs are not displayed on "visualize" tab. Then i realized flow bytes and packets are always "0". 
![image](https://github.com/akvorado/akvorado/assets/76643470/0b7d0c94-a3b9-41a3-bab7-dde22b89f818)

It seems like it's similar to dubefab's case: https://github.com/akvorado/akvorado/discussions/537
I captured packets and analyzed it using wireshark again. Some packets/flow_deleted,unknown.../ have "octet" fields and some/flow_denied.../ doesn't have it.
![image](https://github.com/akvorado/akvorado/assets/76643470/d1ff6f9e-673e-4baa-80c6-d8480c9780f8)
![image](https://github.com/akvorado/akvorado/assets/76643470/ef4ce0e1-c2a7-4539-89fd-966b08a2f8c4)

I guess Akvorado couldn't understand octet information that ASA is sending. What do you think?

**Reply by vincentbernat** [reply]
Yes, it does not. It seems to be a bidirectional flow and it is currently not supported (we need to export two entries from one flow).

**Reply by Captain-E** [reply]
So, there's nothing we can do about it at the moment? am i correct?

**Reply by vincentbernat** [reply, marked as answer]
Correct.

**Reply by Captain-E** [reply]
ok. thank you for response.
I appreciate your hard work :))
